### PR TITLE
Add `--noReconnectAfter 1d` option when available in the agent image.

### PR DESCRIPTION
**Pull request by Vlatombe**
https://github.com/jenkinsci/docker-agent/pull/809 added support for `REMOTING_OPTS` in order to pass custom remoting options to the agent.

This makes use of the new option to pass `--noReconnectAfter 1d`, allowing the agent to bail out after 1 day of not being able to contact the controller.

This can be tweaked using `-Dorg.csanchez.jenkins.plugins.kubernetes.PodTemplateBuilder.noReconnectAfter=30d` using the syntax allowed by the agent per https://github.com/jenkinsci/remoting/pull/738

Tested using `mvn hpi:run -Dorg.csanchez.jenkins.plugins.kubernetes.PodTemplateBuilder.noReconnectAfter=1m`, launching a kubernetes agent, then turning off the controller. The agent bails out after 1 minute in that case.

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by jglick** [comment]
You could just continue to use the existing (now deprecated) variables for existing purposes, and allow the new variable to be set for new purposes such as `-noReconnectAfter`. That would be more compatible. Eventually you would want to do this to clean up.

**Comment by jglick** [comment]
:thinking: 

> set by default for K8s agents, at least if pod GC is enabled

or just a new text field in `KubernetesCloud` for a list of extra options?

**Comment by Vlatombe** [comment]
Simplified the patch accordingly.

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilderTest.java [hunk 1]
```diff
@@ -451,8 +451,8 @@ private void validateContainers(Pod pod, KubernetesSlave slave, boolean directCo
             if ("jnlp".equals(c.getName())) {
                 validateJnlpContainer(c, slave, directConnection);
             } else {
-                List<EnvVar> env = c.getEnv();
-                assertThat(env.stream().map(EnvVar::getName).collect(toList()), everyItem(not(isIn(exclusions))));
+                assertThat(
+                        c.getEnv().stream().map(EnvVar::getName).collect(toList()), everyItem(not(is(in(exclusions)))));
```
Comment (jglick) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilderTest.java:455]: Is this related?

**Review by jglick — APPROVED** [review]
Looks nice and simple.

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilderTest.java (hunk 1 above)
Comment (Vlatombe) [reply src/test/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilderTest.java]: Actually no, reverted.
